enumerable.rb

```diff
@@ -0,0 +1,102 @@
+# Custom Enumerable Methods
+module ::Enumerable
+  # Calls block with the item for each item in enum.
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |n| yield self[n] }
+    self
+  end
+
+  # Loops, Calling The Block With Item And Index
+  def my_each_with_index(&block)
+    for index in 0...count
+      block.call(self[index], index)
+    end
+  end
```

Comment:
### #my_each_with_index

This method should behave the same way as the each but with an additional variable that would be the index. refer to this [doc](https://apidock.com/ruby/v2_5_5/Array/each_index).

right now it's having the same behavior flaw as your #my_each. for some reason, you didn't check here if a block is given at all. Make sure your method work when no block is given as well

Reply:
Fixed in https://github.com/epinczinger/ruby_enumerables/pull/7/commits/33969bc89e0268479e9564169e34d8a72a05476d